gfauto/gfauto/amber_converter.py

```diff
@@ -513,51 +513,62 @@ def get_amber_script_shader_def(shader: Shader, name: str) -> str:
 
     result = get_amber_script_header(amberfy_settings)
 
-    variant = shader_job_amber_test.variant
+    jobs = [shader_job_amber_test.variant]
 
-    variant_vertex_shader_name = f"{variant.name_prefix}_vertex_shader"
-    variant_fragment_shader_name = f"{variant.name_prefix}_fragment_shader"
+    if shader_job_amber_test.reference:
+        jobs.append(shader_job_amber_test.reference)
 
-    # Define shaders.
+    for job in jobs:
+        prefix = job.name_prefix
 
-    result += get_amber_script_shader_def(
-        variant.vertex_shader, variant_vertex_shader_name
-    )
+        vertex_shader_name = f"{prefix}_vertex_shader"
+        fragment_shader_name = f"{prefix}_fragment_shader"
 
-    result += get_amber_script_shader_def(
-        variant.fragment_shader, variant_fragment_shader_name
-    )
+        # Define shaders.
 
-    # Define uniforms for variant shader job.
+        result += get_amber_script_shader_def(
+            job.vertex_shader, vertex_shader_name
+        )
 
-    result += "\n"
-    result += variant.uniform_definitions
+        result += get_amber_script_shader_def(
+            job.fragment_shader, fragment_shader_name
+        )
 
-    # Currently, this buffer must be called "framebuffer" to be able to dump it from Amber.
-    result += "\nBUFFER framebuffer FORMAT B8G8R8A8_UNORM\n"
+        # Define uniforms for variant shader job.
 
-    # Create a pipeline that uses the variant shaders and writes the output to "framebuffer".
+        result += "\n"
+        result += job.uniform_definitions
 
-    result += "\nPIPELINE graphics gfz_pipeline\n"
-    result += f"  ATTACH {variant_vertex_shader_name}\n"
-    result += f"  ATTACH {variant_fragment_shader_name}\n"
-    result += "  FRAMEBUFFER_SIZE 256 256\n"
-    result += "  BIND BUFFER framebuffer AS color LOCATION 0\n"
-    result += variant.uniform_bindings
-    result += "END\n"
-    result += "CLEAR_COLOR gfz_pipeline 0 0 0 255\n"
+        result += f"\nBUFFER {prefix}_framebuffer FORMAT B8G8R8A8_UNORM\n"
 
-    # Run the pipeline.
+        # Create a pipeline.
+
+        result += f"\nPIPELINE graphics {prefix}_pipeline\n"
+        result += f"  ATTACH {vertex_shader_name}\n"
+        result += f"  ATTACH {fragment_shader_name}\n"
+        result += "  FRAMEBUFFER_SIZE 256 256\n"
+        result += f"  BIND BUFFER {prefix}_framebuffer AS color LOCATION 0\n"
+        result += job.uniform_bindings
+        result += "END\n"
+        result += f"CLEAR_COLOR {prefix}_pipeline 0 0 0 255\n"
+
+        # Run the pipeline.
+
+        result += f"\nCLEAR {prefix}_pipeline\n"
+        result += f"RUN {prefix}_pipeline DRAW_RECT POS 0 0 SIZE 256 256\n"
+        result += "\n"
+
+    # Add fuzzy compare of framebuffers if there's more than one pipeline
 
-    result += "\nCLEAR gfz_pipeline\n"
-    result += "RUN gfz_pipeline DRAW_RECT POS 0 0 SIZE 256 256\n"
+    for pipeline_index in range(len(jobs) - 1):
```

Comment:
A bit picky, but I think we would prefer to always compare to the first pipeline (usually the reference, assuming you make the change above). So perhaps do `range(1, len(jobs))` and compare [0] and [pipeline_index]